NuRadioReco/modules/channelTimeOffsetCalculator.py

```diff
@@ -0,0 +1,159 @@
+        self.__medium = None
+        self.__antenna_provider = NuRadioReco.detector.antennapattern.AntennaPatternProvider()
+
+    def begin(self, electric_field_template, medium, use_sim=False):
+        """
+        Setup method for the module
+
+        Parameters:
+        ________________
+        electric_field_template: NuRadioReco.framework.base_trace.BaseTrace object (or child of BaseTrace class)
+            An electric field waveform to be used as a template. It will be folded with the antenna and amplifier
+            response to get a voltage template, which is then used to estimate timing differences between channels.
+        medium: NuRadioMC.utilities.medium.Medium object
+            The index of refraction profile of the ice.
+        use_sim: Boolean (default: False)
+            If true, the simulated vertex is used, otherwise it is assumed that the vertex is stored on either the
+            vertex_2D_fit (for the neutrino2DVertexReconstructor) or the nu_vertex station parameter
+        """
+        self.__use_sim = use_sim
+        self.__electric_field_template = electric_field_template
+        self.__medium = medium
+
+    @register_run()
+    def run(self, event, station, det, channel_ids, passband):
+        """
+        Run the module on a station
+
+        Parameters:
+        _________________
+        event: NuRadioReco.framework.event.Event object
+            The event the module should be run on
+        station: NuRadioReco.framework.station.Station object
+            The station the module should be run on
+        det: NuRadioReco.detector.detector.Detector object of child object
+            The detector description
+        channel_ids: array of int
+            IDs of the channels the module should be run on
+        passband: array of float
+            Lower and upper bound of the bandpass filter that is applied to the
+            channel waveforms when determining correlation to the template. Can
+            be used to filter out frequencies that are dominated by noise.
+        """
+
+        # Create data structured to store pulse properties
+        propagation_times = np.zeros((len(channel_ids), 3))
+        receive_angles = np.zeros((len(channel_ids), 3))
+        found_solutions = np.zeros((len(channel_ids), 3))
+        # Get vertex position
+        vertex_position = None
+        if self.__use_sim:
+            for sim_shower in event.get_sim_showers():
+                if sim_shower.has_parameter(shp.vertex):
+                    vertex_position = sim_shower.get_parameter(shp.vertex)
+                    break
+        else:
+            if station.has_parameter(stnp.nu_vertex):
+                vertex_position = station.get_parameter(stnp.nu_vertex)
+            elif station.has_parameter(stnp.vertex_2D_fit):
+                vertex_2d = station.get_parameter(stnp.vertex_2D_fit)
+                vertex_position = np.array([vertex_2d[0], 0, vertex_2d[1]])
```

Comment:
Why is there a 0 at the position one? probably right though.

Reply:
Yes, I think putting it along the x axis is fine here